students/nskvarch/lesson4/mailroom2.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+# Lesson 4 exercise "the mailroom part 2", created by Niels Skvarch
+
+# Import Modules needed to run
+import sys
+import os
+
+# define global variables
+donor_db = {"Bob Johnson": [3772.32, 512.17],
+            "Fred Billyjoe": [877.33, 455.50, 23.45],
+            "Harry Richard": [1.50],
+            "Old Gregg": [1663.23, 4300.87, 10432.0],
+            "Jerry Vars": [19.95, 653.21, 99.00],
+            }
+
+menu = ("\n".join(("Welcome to the Mailroom.",
+                   "Please choose from below options:",
+                   "1 - Add a new contribution and display sample thank you letter",
+                   "2 - Display a donor Report",
+                   "3 - Create a thank you letter for all donors",
+                   "4 - Exit",
+                   "--> ")))
+
+
+# define functions
+def print_report(data_lines):
+    """Take in a table in the form of a list and display it"""
+    sizes = [0] * len(data_lines[0])
+    for line in data_lines:
+        for item_index, item in enumerate(line):
+            if len(item) > sizes[item_index]:
+                sizes[item_index] = len(item)
+
+    for line in data_lines:
+        print(line[0].ljust(sizes[0]) + "    $    " + line[1].ljust(sizes[1]) + "    " + line[2].rjust(
+            sizes[2]) + "    \t    " + line[3].rjust(sizes[3]))
+
+
+def run_report(donor_db):
+    """take in the donor database, create organize and sort a table in the form of a list and then hand the table off
+    to the printer function """
+    data_lines = []
+    for name in donor_db:
+        data = donor_db[name]
+        data_lines.append((name, str(max(data)), str(len(data)), str(sum(data) / len(data))))
+
+    to_print = [("Name", "Total Donated", "Times Donated", "Average Donation")] + sorted(data_lines, key=lambda x: x[0])
+    print_report(to_print)
+
+
+def thanks(donor_db):
+    """create a thank you note customized to the name provided, 
+    if the name was not in the donor database already, add the name and promt for a donation ammount"""
+    name = input("\nPlease enter the name of the person donating: ")
```

Comment:
This should also give the user the option to get a list of current donors in the database.

Reply:
added the list option to the sub menu for this item